kafka-connect-s3/src/test/java/io/confluent/connect/s3/S3SinkConnectorConfigTest.java

```diff
@@ -115,6 +129,93 @@ public void testAvroDataConfigSupported() throws Exception {
     properties.put(PartitionerConfig.PARTITIONER_CLASS_CONFIG, DefaultPartitioner.class.getName());
     List<ConfigValue> values = S3SinkConnectorConfig.getConfig().validate(properties);
+    assertDefaultPartitionerVisibility(values);
+
+    properties.put(PartitionerConfig.PARTITIONER_CLASS_CONFIG, FieldPartitioner.class.getName());
+    assertFieldPartitionerVisibility();
+
+    properties.put(PartitionerConfig.PARTITIONER_CLASS_CONFIG, DailyPartitioner.class.getName());
+    values = S3SinkConnectorConfig.getConfig().validate(properties);
+    assertTimeBasedPartitionerVisibility(values);
+
+    properties.put(PartitionerConfig.PARTITIONER_CLASS_CONFIG, HourlyPartitioner.class.getName());
+    values = S3SinkConnectorConfig.getConfig().validate(properties);
+    assertTimeBasedPartitionerVisibility(values);
+
+    properties.put(
+        PartitionerConfig.PARTITIONER_CLASS_CONFIG,
+        TimeBasedPartitioner.class.getName()
+    );
+    values = S3SinkConnectorConfig.getConfig().validate(properties);
+    assertNullPartitionerVisibility(values);
+
+    Partitioner<?> klass = new Partitioner<FieldSchema>() {
+      @Override
+      public void configure(Map<String, Object> config) {}
+
+      @Override
+      public String encodePartition(SinkRecord sinkRecord) {
+        return null;
+      }
+
+      @Override
+      public String generatePartitionedPath(String topic, String encodedPartition) {
+        return null;
+      }
+
+      @Override
+      public List<FieldSchema> partitionFields() {
+        return null;
+      }
+    };
+
+    properties.put(
+        PartitionerConfig.PARTITIONER_CLASS_CONFIG,
+        klass.getClass().getName()
+    );
+    values = S3SinkConnectorConfig.getConfig().validate(properties);
+    assertNullPartitionerVisibility(values);
+  }
+
+  @Test
+  public void testConfigurableCredentialProvider() {
+    final String ACCESS_KEY_VALUE = "AKIAAAAAKKKKIIIIAAAA";
+    final String SECRET_KEY_VALUE = "WhoIsJohnGalt?";
+
+    properties.put(
+        S3SinkConnectorConfig.CREDENTIALS_PROVIDER_CLASS_CONFIG,
+        DummyAssertiveCredentialsProvider.class.getName()
+    );
+    String configPrefix = S3SinkConnectorConfig.CREDENTIALS_PROVIDER_CONFIG_PREFIX;
+    properties.put(configPrefix.concat(ACCESS_KEY_NAME), ACCESS_KEY_VALUE);
+    properties.put(configPrefix.concat(SECRET_KEY_NAME), SECRET_KEY_VALUE);
+    properties.put(configPrefix.concat(CONFIGS_NUM_KEY_NAME), "3");
+    connectorConfig = new S3SinkConnectorConfig(properties);
+
+    AWSCredentialsProvider credentialsProvider = connectorConfig.getCredentialsProvider();
+
+    assertEquals(ACCESS_KEY_VALUE, credentialsProvider.getCredentials().getAWSAccessKeyId());
+    assertEquals(SECRET_KEY_VALUE, credentialsProvider.getCredentials().getAWSSecretKey());
+  }
+
+  @Test
+  public void testConfigurableCredentialProviderMissingConfigs() {
+
+    thrown.expect(ConfigException.class);
```

Comment:
usually we add the exception in `@Test` annotation, because checking for a substring in the exception message is usually brittle anyways (Still necessary in some cases).

Reply:
I think it this case it makes sense to check exception's message since `ConfigException` might be thrown in a vast number of cases.

Reply:
not too vast if you are expecting one maybe. Fair enough.